usaspending_api/bulk_download/v2/views.py

```diff
@@ -416,13 +435,33 @@ class BulkDownloadAwardsViewSet(BaseDownloadViewSet):
     def process_filters(self, filters, award_level):
         """Filter function for Bulk Download Award Generation"""
 
-        for required_param in ['award_types', 'agency', 'date_type', 'date_range']:
-            if required_param not in filters:
-                raise InvalidParameterException('{} filter not provided'.format(required_param))
-
         table = value_mappings[award_level]['table']
         queryset = table.objects.all()
 
+        keyword = filters['keyword']
+        if keyword:
+            logger.info('Getting ids based on keyword: {}'.format(keyword))
+            elastic_search_complete = False
+            size = 500000
+            while not elastic_search_complete:
+                try:
+                    transaction_ids = elasticsearch_helper.get_transaction_ids(keyword=keyword, size=size)
+                    # flatten ids
+                    transaction_ids = list(itertools.chain.from_iterable(transaction_ids))
+                    elastic_search_complete = True
+                except (TransportError, ConnectionError) as e:
+                    logger.error(e)
+                    transaction_ids = []
+                    size = size//10
```

Comment:
So if this errors out for any reason (timeout, connection issues but not a generic exception), we'll decrease the request by a factor of 10 and retry. We should make the relevant stake holders aware of this but this is probably a good policy to employ.

Reply:
So this is not decreasing the max size of the transactions, but rather the size of the chunks elastic is returning. So if this errors out, the elastic search will not ask for 500,000 once but in ten requests of 50,000, and so on.